src/rules/selector-no-qualifying-type/index.js

```diff
@@ -0,0 +1,108 @@
+"use strict"
+
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-no-qualifying-type"
+export const messages = ruleMessages(ruleName, {
+  rejected: "Unexpected selector qualified by type",
+})
+
+export default (enabled, options) => {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: enabled,
+      possible: [ true, false ],
+    }, {
+      actual: options,
+      possible: {
+        ignore: [ "attribute", "class", "id" ],
+      },
+      optional: true,
+    })
+
+    const isElementSelector = (selector) => {
+      const firstChar = selector.trim().substring(0, 1)
+      return /^[a-z]+$/i.test(firstChar)
+    }
+
+    const hasAttributeQualifiedByElement = (selector) => {
+      const selectorArray = selector.split(" ")
```

Comment:
FYI, if postcss-selector-parser doesn't work, a split array of selectors is already available via PostCSS at `ruleNode.selectors` (plural).